src/main/tray.ts

```diff
@@ -0,0 +1,67 @@
+import { app as rawApp, ipcMain, Menu, Tray } from 'electron'
+import { globalShortcut } from 'electron'
+import path from 'path'
+import * as mainWindow from './windows/main'
+import { ExtendedAppMainProcess } from './types'
+import { getLogger } from '../shared/logger'
+
+let tray: Tray = null
+
+const app = rawApp as ExtendedAppMainProcess
+const log = getLogger('main/tray')
+
+ipcMain.on('updateTrayIcon', updateTrayIcon)
+
+export function updateTrayIcon() {
+  const tx = app.translate
+
+  const IsMac = process.platform === 'darwin'
+  if (!IsMac) {
+    if (app.state.saved.minimizeToTray !== true) {
+      // User doesn't want tray icon => destroy it
+      if (tray != null) {
+        log.info('destroy icon tray')
+        tray.destroy()
+      }
+    } else {
+      // Add tray icon
+      log.info('add icon tray')
+      tray = new Tray(
+        `${path.join(__dirname, '..', '..', 'images', 'deltachat.ico')}`
```

Comment:
rather use `appIcon()` here
to import it:
```js
import { appIcon } from './application-constants'
```
and extend the function in the application constants to return the path to the ico file instead of the png on windows. (https://nodejs.org/api/process.html#process_process_platform)
```